voxel_isosurface_tool.cpp

```diff
@@ -33,7 +33,7 @@ namespace {
 
 inline void do_op(VoxelBuffer &buffer, int x, int y, int z, float d1, VoxelIsoSurfaceTool::Operation op) {
 
-	float res;
+	float res=0;
```

Comment:
Does GCC still complain if instead of giving a default value, you add a `default` label in the switch?

Reply:
Yes. The default section didn't affect the warnings, which come later on:

We get four of these lovely warnings:

```
In file included from modules/voxel/voxel_isosurface_tool.h:4,
                 from modules/voxel/voxel_isosurface_tool.cpp:1:
modules/voxel/voxel_buffer.h: In member function 'void VoxelIsoSurfaceTool::do_heightmap(Ref<Image>, Vector3, real_t, VoxelIsoSurfaceTool::Operation)':
modules/voxel/voxel_buffer.h:41:34: warning: 'res' may be used uninitialized in this function [-Wmaybe-uninitialized]
   int v = static_cast<int>(128.f * iso + 128.f);
                            ~~~~~~^~~~~
modules/voxel/voxel_isosurface_tool.cpp:36:8: note: 'res' was declared here
  float res;
        ^~~
```

Reply:
the thing is, GCC is completely wrong about that :p `res` would be initialized in all `case` labels, AND the `default` anyways. A remaining case is if an exception is thrown, which is turned off in Godot. It worries me a little because this function can run a very high number of times.

I guess you can initialize for now, tbh even the `MIN` and `MAX` macros are bad for perf here, because they expand into more function calls than necessary :p

Reply:
I was confused about what I was supposed to test. The warning remained with an empty default. However if I initialize to 0 in default case, the warning goes away. PR updated